src/main/java/com/contentful/java/cma/model/CMASpaceMembership.java

```diff
@@ -0,0 +1,196 @@
+package com.contentful.java.cma.model;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+/**
+ * Model class, exposing a space membership.
+ */
+public class CMASpaceMembership extends CMAResource {
+  private boolean admin;
+  private CMALink user;
+  private List<CMALink> roles;
+  private String email;
+
+  /**
+   * Create a new membership.
+   */
+  public CMASpaceMembership() {
+    super(CMAType.SpaceMembership);
+  }
+
+  /**
+   * Update admin flag.
+   * <p>
+   * If you set this flag to false, you have to provide some roles this membership should be part
+   * of.
+   *
+   * @param admin true, if this membership should be an admin.
+   * @return this membership for ease of chaining.
+   * @see #setRoles(CMALink...)
+   * @see #addRole(CMALink)
+   */
+  public CMASpaceMembership setAdmin(boolean admin) {
+    this.admin = admin;
+    return this;
+  }
+
+  /**
+   * @return a link to the user of this membership
+   */
+  public CMALink getUser() {
+    return user;
+  }
+
+  /**
+   * Update or set the user for this membership.
+   *
+   * @param user which user should be part of this membership?
+   * @return this membership for ease of chaining.
+   */
+  public CMASpaceMembership setUser(CMALink user) {
+    this.user = user;
+    return this;
+  }
+
+  /**
+   * @return a list of links to the roles this membership contains.
+   */
+  public List<CMALink> getRoles() {
+    return roles;
+  }
+
+  /**
+   * Replace all roles with the given argument.
+   *
+   * @param roles the roles to be used.
+   * @return this instance of membership for ease of chaining.
+   * @throws IllegalArgumentException if roles is null.
+   * @throws IllegalArgumentException if roles does not contain any elements.
+   */
+  public CMASpaceMembership setRoles(CMALink... roles) {
```

Comment:
[Varargs](https://docs.oracle.com/javase/1.5.0/docs/guide/language/varargs.html) in java